stdlib/2and3/wsgiref/types.pyi

```diff
@@ -15,27 +15,34 @@
 # you need to use 'WSGIApplication' and not simply WSGIApplication when type
 # hinting your code.  Otherwise Python will raise NameErrors.
 
-import sys
-from typing import Callable, Dict, Iterable, List, Optional, Tuple, Type, Union, Any
+from typing import Callable, Dict, Iterable, List, Optional, Tuple, Type, Union, Any, Text, Protocol
 from types import TracebackType
 
 _exc_info = Tuple[Optional[Type[BaseException]],
                   Optional[BaseException],
                   Optional[TracebackType]]
-if sys.version_info < (3,):
-    _Text = Union[unicode, str]
-    _BText = _Text
-else:
-    _Text = str
-    _BText = Union[bytes, str]
-WSGIEnvironment = Dict[_Text, Any]
+_BText = Union[bytes, str]
+WSGIEnvironment = Dict[Text, Any]
 WSGIApplication = Callable[
     [
         WSGIEnvironment,
         Union[
-            Callable[[_Text, List[Tuple[_Text, _Text]]], Callable[[_BText], None]],
-            Callable[[_Text, List[Tuple[_Text, _Text]], _exc_info], Callable[[_BText], None]]
+            Callable[[Text, List[Tuple[Text, Text]]], Callable[[_BText], None]],
+            Callable[[Text, List[Tuple[Text, Text]], _exc_info], Callable[[_BText], None]]
         ]
     ],
-    Iterable[_BText]
+    Iterable[bytes]
 ]
+
+# WSGI input streams per PEP 3333
+class InputStream(Protocol):
+    def read(self, size: int = ...) -> bytes: ...
+    def readline(self, size: int = ...) -> bytes: ...
+    def readlines(self, hint: int = ...) -> List[bytes]: ...
+    def __iter__(self) -> Iterable[bytes]: ...
+
+# WSGI error streams per PEP 3333
+class ErrorStream(Protocol):
+    def flush(self) -> None: ...
+    def write(self, s: str) -> None: ...
+    def writelines(self, seq: List[str]) -> None: ...
```

Comment:
Not `Iterable`?

Reply:
Good question. Unfortunately, PEP 3333 is not explicit about the expected type. https://docs.python.org/3/library/io.html says about `IOBase.writelines()`: "Write a list of lines to the stream." I'd say it's safer to use `List` (or at least `Sequence`) here, in case some WSGI server really expects a list and does not work with iterables.